solr/core/src/test/org/apache/solr/metrics/DelegateRegistryTimerTest.java

```diff
@@ -0,0 +1,218 @@
+    assertEquals(1, delegateRegistryTimer.getDelegateTimer().getCount());
+    assertEquals(
+        10000000000.0, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMean(), 0.0);
+    assertEquals(
+        10000000000.0, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMedian(), 0.0);
+    assertEquals(10000000000L, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMax());
+
+    delegateRegistryTimer.update(20, TimeUnit.SECONDS);
+    delegateRegistryTimer.update(30, TimeUnit.SECONDS);
+    assertEquals(3, delegateRegistryTimer.getPrimaryTimer().getCount());
+    assertEquals(
+        20000000000.0, delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMean(), 0.0);
+    assertEquals(
+        20000000000.0, delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMedian(), 0.0);
+    assertEquals(30000000000L, delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMax());
+    assertEquals(20000000000.0, delegateRegistryTimer.getSnapshot().getMean(), 0.0);
+    assertEquals(20000000000.0, delegateRegistryTimer.getSnapshot().getMedian(), 0.0);
+    assertEquals(30000000000L, delegateRegistryTimer.getSnapshot().getMax());
+    assertEquals(3, delegateRegistryTimer.getDelegateTimer().getCount());
+    assertEquals(
+        20000000000.0, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMean(), 0.0);
+    assertEquals(
+        20000000000.0, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMedian(), 0.0);
+    assertEquals(30000000000L, delegateRegistryTimer.getDelegateTimer().getSnapshot().getMax());
+  }
+
+  @Test
+  public void timeContext() throws InterruptedException {
+    DelegateRegistryTimer delegateRegistryTimer =
+        new DelegateRegistryTimer(
+            Clock.defaultClock(), timerSupplier.newMetric(), timerSupplier.newMetric());
+    Timer.Context time = delegateRegistryTimer.time();
+    Thread.sleep(100);
+    time.close();
+    assertTrue(delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMean() > 100000);
+    assertTrue(delegateRegistryTimer.getDelegateTimer().getSnapshot().getMean() > 100000);
+    assertEquals(
+        delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMean(),
+        delegateRegistryTimer.getDelegateTimer().getSnapshot().getMean(),
+        0.0);
+    assertEquals(
+        delegateRegistryTimer.getPrimaryTimer().getSnapshot().getMean(),
+        delegateRegistryTimer.getSnapshot().getMean(),
+        0.0);
+  }
+
+  @Test
+  public void timeSupplier() {
+    DelegateRegistryTimer delegateRegistryTimer =
+        new DelegateRegistryTimer(
+            Clock.defaultClock(), timerSupplier.newMetric(), timerSupplier.newMetric());
+    Long supplierResult =
+        delegateRegistryTimer.timeSupplier(
+            () -> {
+              try {
+                Thread.sleep(100);
+              } catch (InterruptedException e) {
+                Assert.fail("Thread was interrupted while sleeping");
```

Comment:
Would need to double check that this will actually fail the test instead of getting caught in some other exception handler if it goes to a different thread.

Reply:
I took a different approach here that should just avoid that potential.